File: src/main/java/org/athenian/ContainsLetter.java
package org.athenian;

public class ContainsLetter {

    public static boolean containsLetterLoop(char c, String s) {
        String cstr = Character.toString(c);

        for (int i = 0; i < s.length(); i++)
            if (s.substring(i, i + 1).equals(cstr))
                return true;

        return false;
    }

    public static boolean containsLetterRecursion(char c, String s) {
        if (s.length() == 0)
            return false;

        String firstLetter = s.substring(0, 1);
        if (firstLetter.equals(Character.toString(c)))
            return true;

        return containsLetterRecursion(c, s.substring(1));
    }

    public static boolean containsLetterStream(char c, String s) {
        return s.chars()
                .filter(value -> value == c)
                .findAny()
                .isPresent();
    }
}

File: src/main/kotlin/org/athenian/ContainsLetter.kt
package org.athenian

fun containsLetterLoop(c: Char, s: String) =
    when {
        s.isEmpty() -> false
        else -> {
            repeat(s.length) {
                if (s[it] == c)
                    return true
            }
            false
        }
    }

fun containsLetterRecursion(c: Char, s: String): Boolean =
    when {
        s.isEmpty() -> false
        s[0] == c -> true
        else -> containsLetterRecursion(c, s.substring(1))
    }

fun containsLetterStream(c: Char, s: String) = s.any { value -> value == c }
